java/code/src/com/suse/scc/SCCSystemRegistrationManager.java

```diff
@@ -0,0 +1,170 @@
+                        getPayload(cacheItem),
+                        itemCredentials.getUsername(),
+                        itemCredentials.getPassword());
+                cacheItem.setSccId(systemCredentials.getId());
+                cacheItem.setSccLogin(systemCredentials.getLogin());
+                cacheItem.setSccPasswd(systemCredentials.getPassword());
+                cacheItem.setSccRegistrationRequired(false);
+                cacheItem.setRegistrationErrorTime(null);
+                cacheItem.setCredentials(itemCredentials);
+            }
+            catch (SCCClientException e) {
+                LOG.error("Error registering system " + cacheItem.getId(), e);
+                cacheItem.setRegistrationErrorTime(new Date());
+            }
+            cacheItem.getOptServer().ifPresent(ServerFactory::save);
+        });
+    }
+
+    private SCCRegisterSystemJson getPayload(SCCRegCacheItem rci) {
+        Server srv = rci.getOptServer().get();
+        List<SCCMinProductJson> products = Opt.fold(srv.getInstalledProductSet(),
+                () -> {
+                    return new LinkedList<SUSEProduct>();
+                },
+                s -> {
+                    List<SUSEProduct> prd = new LinkedList<>();
+                    prd.add(s.getBaseProduct());
+                    prd.addAll(s.getAddonProducts());
+                    return prd;
+                }
+        ).stream()
+                .map(p -> new SCCMinProductJson(p))
+                .collect(Collectors.toList());
+
+        Map<String, String> hwinfo = new HashMap<>();
+        Optional<CPU> cpu = ofNullable(srv.getCpu());
+        cpu.flatMap(c -> ofNullable(c.getNrCPU())).ifPresent(c -> hwinfo.put("cpus", c.toString()));
+        cpu.flatMap(c -> ofNullable(c.getNrsocket())).ifPresent(c -> hwinfo.put("sockets", c.toString()));
+        hwinfo.put("arch", srv.getServerArch().getLabel().split("-")[0]);
+        if (srv.isVirtualGuest()) {
+            hwinfo.put("hypervisor", srv.getVirtualInstance().getType().getHypervisor().orElse(""));
+            hwinfo.put("cloud_provider", srv.getVirtualInstance().getType().getCloudProvider().orElse(""));
+            ofNullable(srv.getVirtualInstance().getUuid()).ifPresent(u -> {
+                hwinfo.put("uuid", u.replaceAll("(\\w{8})(\\w{4})(\\w{4})(\\w{4})(\\w{12})",
+                        "$1-$2-$3-$4-$5"));
+            });
+        }
+        else {
+            // null == physical instance
+            hwinfo.put("hypervisor", null);
+        }
+
+        String login = rci.getOptSccLogin().orElseGet(() -> {
+            String l = String.format("%s-%s", ContentSyncManager.getUUID(), srv.getId().toString());
+            rci.setSccLogin(l);
+            SCCCachingFactory.saveRegCacheItem(rci);
+            return l;
+        });
+        String passwd = rci.getOptSccPasswd().orElseGet(() -> {
+            String pw = RandomStringUtils.randomAlphanumeric(64);
```

Comment:
Not sure if its necessary but we should consider using `SecureRandom` for things like passwords.

Reply:
I think it is not needed here. But when we generate the secret of a system it might be a good idea to change it.